numpy/core/records.py

```diff
@@ -238,17 +237,14 @@ def __getattribute__(self, attr):
         res = fielddict.get(attr, None)
         if res:
             obj = self.getfield(*res[:2])
-            # if it has fields return a recarray,
-            # if it's a string ('SU') return a chararray
+            # if it has fields return a record,
             # otherwise return the object
             try:
                 dt = obj.dtype
             except AttributeError:
-                return obj
+                return obj #happens if field is Object type
             if dt.fields:
-                return obj.view(obj.__class__)
-            if dt.char in 'SU':
-                return obj.view(chararray)
+                return obj.view((record, obj.dtype.descr))
```

Comment:
Oh, and to clarify what this line does: It makes nested types behave like records too.

Demostration:

    >>> rec = np.rec.array([('abc ', (1,1), 1), ('abc', (2,3), 1)],
                                          dtype=[('foo', 'S4'), ('bar', [('A', int), ('B', int)]), ('baz', int)])

Old behavior:

    >>> rec[0].bar.A
    AttributeError: 'numpy.void' object has no attribute 'A'

New behavior:

    >>> rec[0].bar.A
    1